src/components/Reown/modals/BaseNanoContractModal.js

```diff
@@ -13,8 +13,22 @@ import helpers from '../../../utils/helpers';
 import { NanoContractActions } from '../NanoContractActions';
 import AddressList from '../../AddressList';
 import { NANO_UPDATE_ADDRESS_LIST_COUNT } from '../../../constants';
-import { constants, nanoUtils } from '@hathor/wallet-lib';
+import { constants, nanoUtils, numberUtils, dateUtils, scriptsUtils, bufferUtils } from '@hathor/wallet-lib';
 import { DAppInfo } from '../DAppInfo';
+import { SignedDataDisplay } from '../SignedDataDisplay';
+
+/**
+ * Parse script data from hex string
+ */
+const parseScriptData = (scriptData, network) => {
+  try {
+    const script = bufferUtils.hexToBuffer(scriptData);
+    return scriptsUtils.parseScript(script, network);
```

Comment:
suggestion(non-blocking): We could reproduce the behavior from the `TxData` file, which adds a bit of error context.

https://github.com/HathorNetwork/hathor-wallet/blob/65a57c1ae0f680decc54c15418385b27c43da7a7/src/components/TxData.js#L555

Reply:
This script doesn't need to be decoded. We try to decode but it can be literally anything. Having a message like this seems an error, which is not.